### ISSUE TITLE
Use sub Emitter for PushButtonIO 

### ISSUE THREAD

**Issue by zepumph** [phetsims/axon#180]
Like NodeIO having sub Properties, we should use the same pattern for PushButtonIO to handle its "fired" event. With emitter we can trigger it and add a listener.

@chrisklus @samreid and I worked on this briefly today, and I will commit what we have so far.

With this pattern it is very important to have guards to make sure that we aren't reentrant into the emit method, this guard is implemented within the setter functions for NodeIO and TextIO's sub properties, but for sub Emitters, it didn't seem as clean, so we added the guard naively in PushButtonIO's constructor. Things seem to work well, and unit/fuzz tests are passing, but I'm sceptical that this is production level code. @samreid let's not let this issue fall off radar, marking as high priority.

This should be completed before FL stable release.

**Comment by zepumph** [comment]
On top of the above note, we had to manually set this sub Emitter to be `phetioState: false` and I'm not sure why, I just listened to @samreid's recommendation. We should discuss that further.

**Comment by zepumph** [comment]
Also it should be recognized that there is a big todo in `createInstanceDiv` to add an `EmitterIO` creator instead of using `PushButtonIO`'s.

**Comment by zepumph** [comment]
@chrisklus and I worked on this today, we instrumented the emitter in `PushButtonModel`, and deleted the added emitted in `PushButtonIO`.

@samreid this seems like a very nice approach. We want to run it by you, but I think this is a good way to go.

Here is what the data stream currently looks like with the scenery input event as well:
```
584 phetio inputEvent {"point":{"x":557.0909423828125,"y":490.90911865234375},"event":{"pointerId":1,"pointerType":"mouse","clientX":557.0909423828125,"clientY":490.90911865234375,"button":0,"which":1},"command":"mouseUp"}
585 faradaysLaw.faradaysLawScreen.view.flipMagnetButton.pressListener released
	586 faradaysLaw.faradaysLawScreen.view.flipMagnetButton fired
		587 faradaysLaw.faradaysLawScreen.view.flipMagnetButton.fireEmitter emitted
 			588 faradaysLaw.faradaysLawScreen.model.magnet.orientationProperty changed {"oldValue":"NS","newValue":"SN"}
```

Right now we like the way this looks, it is nice and verbose, but also clear. I'm worried a bit about how robust it is to api changes though.

**Comment by zepumph** [comment]
Assigning over to @samreid for review.

**Comment by samreid** [comment]
The part I'm confused about is why we have both the "button fired" and "button firedEmitter emitted".  Were we hoping to eliminate the former?  Perhaps we should chat at next dev meeting.

**Comment by zepumph** [comment]
@samreid @chrisklus and I worked on this today, and we converted all *ButtonModel.js types to use instrumented emitters for their phet-io event triggering events. This worked very well save two bugs we found:
https://github.com/phetsims/sun/issues/372 and https://github.com/phetsims/tandem/issues/60

This issue focuses around adding emitters to the button model types. @samreid please review.

**Comment by samreid** [comment]
@zepumph will delete PushButtonIO.  We don't need emit1/emit2 in the studio interface.  We will add emit1, emit2, etc to EmitterIO once they are first needed.

@zepumph volunteered to look through and remove any types with `events: [ 'fired' ]` that are no longer necessary, and create a new issue to convert remaining types with `events: [ 'fired' ]` to use `Emitter` as appropriate.
